context/app/static/js/components/detailPage/DatasetRelationships/nodeTypes.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { PropsWithChildren } from 'react';
+import { Node, NodeProps, Handle, Position, NodeTypes } from '@xyflow/react';
+import Stack from '@mui/material/Stack';
+import { DatasetIcon } from 'js/shared-styles/icons';
+import Typography from '@mui/material/Typography';
+import { AccountTreeRounded, ExtensionRounded, SvgIconComponent } from '@mui/icons-material';
+import Skeleton from '@mui/material/Skeleton';
+import { Box } from '@mui/system';
+import StatusIcon from '../StatusIcon';
+import { usePipelineInfo } from './hooks';
+
+interface CommonNodeInfo extends Record<string, unknown> {
+  name: string;
+}
+
+interface PipelineNodeInfo extends CommonNodeInfo {
+  childDatasets: string[];
+  singleAssay?: boolean;
+  description?: string;
+}
+
+interface DatasetNodeInfo extends CommonNodeInfo {
+  datasetType?: string;
+  status: string;
+}
+
+interface NodeTemplateProps extends PropsWithChildren, CommonNodeInfo {
+  icon?: SvgIconComponent;
+  status?: string;
+  target?: boolean;
+  source?: boolean;
+  rounded?: boolean;
+  bgColor?: string;
+  isLoading?: boolean;
+  href?: string;
+}
+
+export const nodeColors = {
+  primaryDataset: '#F0F3EB',
+  processedDataset: '#EAF0F8',
+  pipeline: '#EFEFEF',
+  componentDataset: '#FBEEEB',
+};
+
+export const nodeIcons = {
+  primaryDataset: DatasetIcon,
+  processedDataset: AccountTreeRounded,
+  pipeline: null,
+  componentDataset: ExtensionRounded,
+};
+
+const nodeHeightRem = 4.125;
+export const nodeHeight = nodeHeightRem * 16;
+
+function NodeTemplate({
+  icon: Icon,
+  status,
+  children,
+  name,
+  target,
+  source,
+  rounded,
+  bgColor,
+  isLoading,
+  href,
+}: NodeTemplateProps) {
+  // Outer wrapper Box makes sure that nodes are always the same height
+  const contents = (
+    <Box height="4.125rem" display="flex" alignItems="center">
+      <Stack
+        direction="column"
+        px={2}
+        py={1}
+        borderRadius={rounded ? '1rem' : 0}
+        maxWidth="18rem"
```

Comment:
Why these `rem` values?

Reply:
Mainly to match the measurements from the Figma designs; the outer `Box` is necessary to make the node positions line up (i.e. by making them all the same height without affecting the appearance of the content)